src/util/metrics.ts

```diff
@@ -0,0 +1,59 @@
+import express from 'express';
+import { Application, Request, Response } from 'express';
+import client from 'prom-client';
+
+import { Log } from '../logging/Log';
+
+const { logger } = Log;
+
+const register = new client.Registry();
+
+export const httpErrorCounter = new client.Counter({
+	name: 'sas_http_errors',
+	help: 'Number of HTTP Errors',
+});
+
+register.registerMetric(httpErrorCounter);
+
+client.collectDefaultMetrics({ register, prefix: 'sas_' });
+
+interface IAppConfiguration {
+	port: number;
+	host: string;
+}
+
+export default class Metrics_App {
+	private app: Application;
+	private readonly port: number;
+	private readonly host: string;
+
+	/**
+	 * @param appConfig configuration for app.
+	 */
+	constructor({ host }: IAppConfiguration) {
+		this.app = express();
+		this.port = 9100;
+		this.host = host;
+
+		this.metricsEndpoint();
+	}
+
+	listen(): void {
+		this.app.listen(this.port, this.host, () => {
+			logger.info(
+				`Metrics Server started at http://${this.host}:${this.port}/`
+			);
+		});
+	}
+
+	/**
+	 * Mount the metrics endpoint.
+	 */
+	private metricsEndpoint() {
+		// Set up the metrics endpoint
+		this.app.get('/metrics', async (_req: Request, res: Response) => {
```

Comment:
Looks like you just need to add the expected return value here.

Reply:
Thanx for the review @CurlyBracketEffect ! 🙏 I added a return value but then I had another lint error. 

What I did now to remove the following lint errors
```
Error: 54:28 error Promise returned in function argument where a void return was expected @typescript-eslint/no-misused-promises
```
and
```
60:4  error  Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator  @typescript-eslint/no-floating-promises
```
 was to : 
- use an async [IIFE](https://developer.mozilla.org/en-US/docs/Glossary/IIFE) inside the callback
- add the `void` operator 

and it works but I am not sure if this is a good practice though 🤔